x-pack/legacy/plugins/ml/public/data_frame/pages/data_frame_new_pivot/components/source_index_preview/use_source_index_data.ts

```diff
@@ -39,7 +39,7 @@ export interface UseSourceIndexDataReturnType {
 }
 
 export const useSourceIndexData = (
-  indexPattern: IndexPattern,
+  indexPattern: IndexPattern | StaticIndexPattern,
```

Comment:
Why is `StaticIndexPattern` needed here? (connected to understanding the difference between the two).

Reply:
without it, TS would fail, but upon further research discovered that this works without `IndexPattern`